crates/db_views/src/post_view.rs

```diff
@@ -324,18 +324,21 @@ impl<'a> PostQuery<'a> {
         }
       }
     }
-
-    if let Some(community_id) = self.community_id {
+    if self.community_id.is_none() && self.community_actor_id.is_none() {
+      query = query
+        .then_order_by(post_aggregates::featured_local.desc());
```

Comment:
Hopefully its just as easy as this. The indexes might need to be updated to make this fast.